File: backend/app/schemas/meta.py
"""基础字典模型（过敏源 / 饮食偏好）"""
from pydantic import BaseModel


class AllergenOut(BaseModel):
    id: int
    name: str
    category: str = ""
    keywords: list[str] = []
    description: str = ""


class DietOut(BaseModel):
    id: int
    name: str
    keywords: list[str] = []
    description: str = ""

File: frontend/src/api/meta.ts
import request from './request'

export interface Allergen {
  id: number
  name: string
  category: string
  keywords: string[]
  description: string
}

export interface DietPreference {
  id: number
  name: string
  keywords: string[]
  description: string
}

export const getAllergensApi = () => request.get<Allergen[]>('/meta/allergens').then((r) => r.data)

export const getDietPreferencesApi = () =>
  request.get<DietPreference[]>('/meta/diet-preferences').then((r) => r.data)
